tools/shared/src/main/scala/org/scalajs/core/tools/optimizer/OptimizerCore.scala

```diff
@@ -2183,6 +2195,21 @@ private[optimizer] abstract class OptimizerCore(
           case _               => default
         }
 
+      case String_+ =>
+        (lhs, rhs) match {
+          case (_, ClassOf(_)) | (ClassOf(_), _) => default
+
+          case (l: Literal, r: Literal) =>
+            StringLiteral(literalToString(l) + literalToString(r))
+          case (t1, t2 @ StringLiteral("")) => foldBinaryOp(op, t2, t1)
+          case (StringLiteral(""), t2) if t2.tpe == StringType => t2
```

Comment:
Try to stay consistent over the various cases, and put the rhs of the cases on the next line as well.